src/sqlshim.php

```diff
@@ -248,13 +248,38 @@ private static function calc_prec_scale( $prec, $scale )
   private static function typify( $row )
   {
     $i = 0;
-    foreach ( $row as $col=>&$val )
-    {
-      $val = self::guesstype($val);
-      $i++;
+    foreach ($row as &$value) {
+        //DBLIB database driver returns everything as strings, so this converts num's back to the correct data type
+        $value = self::convertDataType($value);
     }
     return $row;
   }
+  
+  private static function convertDataType($string){
```

Comment:
You're 100% right! I just rewrote this because yours wasn't working at all for me for some reason. I'm not sure a better solution for this though.